### DISCUSSION TITLE
"connection lost"

### DISCUSSION THREAD

**Discussion by cyrrelsneer** [mbround18/valheim-docker#178] [Help!] [answered]
Hi there,

there was a new update on the Valheim client today coming from Steam and I think ever since then, I cannot connect to the server.
It is probably due to the server running 0.145.6 and the clients version is now 0.146.8, or at least that's my best guess.

Can you confirm this? Is this expected behavior? In case it is, I am going to remove the autoupdate feature in Steam and will first update after your server is being updated.

By the way -thanks so much for this container, highly appreciated! Runs like a charm (before that update ;-) )

**Comment by Henchway** [comment]
Yep, we're experiencing the same issue.

**Comment by prthlr** [comment]
same here!

**Comment by cyrrelsneer** [comment]
Thanks for the confirmation.

Just out of curiosity - do you know how I could continue playing with the backup files that are being created?

**Reply by Henchway** [reply]
I'm guessing you can just extract the information from the container that you initially put into it:

![grafik](https://user-images.githubusercontent.com/37119321/109159092-81a64000-7774-11eb-93ad-b0ad4f09be30.png)

If you put those files into the worlds folder on your pc, you can play locally, if that's what you're referring to.
Btw. where did you find the server-version that's running? I just can't find it.

**Reply by cyrrelsneer** [reply]
ah okay, maybe I'll try it, but I am using it as multiplayer so not sure what to do out of it. For now I'll just wait ;-)

The running version shows up when you start the Valheim client, search for your server and in the list it will show the version of the server.

**Reply by cyrrelsneer** [reply]
so, just downloaded the backup (which contains the whole server, that explains why it's like 600MB in size, maybe I'll deactivate the autobackup feature or so, not sure yet what to do there).
Well the map was like 12mb or so in size and works like a charm being importet into Windows locally.
Would be great to see if I can do changes, upload again into the docker container and check if the changes are there, maybe I'll try that one later as well ;)

**Reply by Henchway** [reply]
I've got it back up, i had to follow these steps:
https://github.com/mbround18/valheim-docker/discussions/125#discussioncomment-378682

However, once i did that i had a new seed, i'm guessing it doesn't load the old savegame directly. I then reinitialized the container and everything was back to normal. This updated the server version:

![grafik](https://user-images.githubusercontent.com/37119321/109164687-009e7700-777b-11eb-949d-511733a7829b.png)

**Reply by cyrrelsneer** [reply]
Thanks - I just did something else to make it up and running which is a bit easier I guess.

I have recreated the container and changed the autoupdate value to now+5 minutes (I am GMT+2, so I updated at 3pm and 10min):

AUTO_UPDATE_SCHEDULE
10 15 * * *

**Reply by Henchway** [reply]
Nice, i assumed that it would auto_update at startup anyway. So far i was a bit unsure whether the auto_update was referring to the docker image or the game.

**Comment by HeadcaseFargone** [comment]
I was able to connect to my world again through Steam after running the auto_update.sh script, but my world was gone.  It was a completely new one.  Not sure how to restore my old one, though I have multiple backups of the server and world volumes.

**Reply by cyrrelsneer** [reply]
hm I didn't have any problems with that, my world is still working.

But - good news is you have a backup ;-)

I would just try the following - copy the 4 files (yourworld.db, yourworld.db.old, yourworld.fwl, yourworld.fwl.old) into the world -  folder, make sure ( I guess? not sure though) that in the parameters worlds name etc. will stay as previously chosen) and fire up the server again...

It's just a best guess, but you can give it a shot!

**Reply by HeadcaseFargone** [reply]
Could that "missing world.dat" line be the culprit?

![image](https://user-images.githubusercontent.com/8780048/109167713-ab933e80-7743-11eb-8c55-a61b223b7761.png)

**Reply by HeadcaseFargone** [reply]
Actually, looks like my .db file is way smaller now.  So I stopped the container, copied over my backup world files, then started again.  Now it doesn't get past DungeonDB Start.  I never get to Game Server connected now.

**Reply by cyrrelsneer** [reply]
maybe, I don't have that specific error in my logs

**Reply by prthlr** [reply]
where can i find the auto_update.sh file?

**Reply by cyrrelsneer** [reply]
my .db files are around 12mb in size.

you can read through the thread above - I have copied it locally and it works, you could verify that for you as well.
But other than that I am also out of options, maybe someone else could step in

**Reply by HeadcaseFargone** [reply]
> where can i find the auto_update.sh file?

docker exec -it ContainerName "/bin/bash"
/home/steam/scripts/auto_update.sh

**Reply by prthlr** [reply]
same problem here @HeadcaseFargone 
after reconnecting to the server, the whole map is new.

**Reply by HeadcaseFargone** [reply]
> same problem here @HeadcaseFargone
> after reconnecting to the server, the whole map is new.

Okay after running auto_update.sh I had to stop my container, overwrite the files in the world volume with the ones from my backup, restart the container, and then wait about 10 whole minutes before it would respond.  It never did show the Game Server connected message, but I was finally able to connect with my old world intact.  Phew.

**Reply by KyneSilverhide** [reply]
Yep, same here. I ran the auto_update.sh script manually, and my entire world was gone... but fortunately I had backups and eevrything seems to be back to normal after moving them back to the docker volume.

I guess the next docker container will include the update, but now I'm a bit scared regarding the AUTO_UPDATE feature...

**Comment by Devidian** [comment]
Same for me, additionally it seems that if i restart one container (i run 2 servers without problems until now) the other gets broken (game crash but container runs with only the log tail command) But even only one container running does not work.

**Comment by TaylorTWBrown** [comment, marked as answer]
I had the same issue.

I resolved it by emptying (deleting and recreating) the SteamCMD and server docker volumes. The components were redownloaded and my world came back as expected.

**Reply by Devidian** [reply]
worked for me too

**Reply by npanagop** [reply]
Worked for me too. More specifically I deleted `valheim/server/` and then run `docker-compose up --detach --force-recreate`. After a while the server was up and my world loaded normally.
